charts/consul/templates/api-gateway-gatewayclassconfig.yaml

```diff
@@ -28,6 +28,18 @@ spec:
     {{- else }}
       http: 8500
     {{- end }}
+  {{- if .Values.apiGateway.managedGatewayClass.deployment }}
+  deployment:
+    {{- if .Values.apiGateway.managedGatewayClass.deployment.defaultInstances }}
+    defaultInstances: {{ .Values.apiGateway.managedGatewayClass.deployment.defaultInstances }}
+    {{- end }}
+    {{- if .Values.apiGateway.managedGatewayClass.deployment.maxInstances }}
+    maxInstances: {{ .Values.apiGateway.managedGatewayClass.deployment.maxInstances }}
+    {{- end }}
+    {{- if .Values.apiGateway.managedGatewayClass.deployment.minInstances }}
+    minInstances: {{ .Values.apiGateway.managedGatewayClass.deployment.minInstances }}
+    {{- end }}
```

Comment:
you can simplify this since these values are not explicitly provided in via helm values like so:

```
{{- with .Values.apiGateway.managedGatewayClass.deployment }}
deployment: 
  {{- toYaml . | nindent 4 }}
{{- end }}
```